libafl/src/executors/timeout.rs

```diff
@@ -322,11 +383,45 @@ where
     }
 }
 
+#[cfg(all(unix, not(target_vendor = "apple")))]
+impl<E> HasOnCrashReset for TimeoutExecutor<E> {
+    /// Disarm currently running timer
+    fn reset_on_crash(&self) {
+        unsafe {
+            let disarmed: libc::itimerspec = zeroed();
+            libc::timer_settime(self.timerid, 0, &disarmed as *const _, null_mut());
+        }
+    }
+}
+
+#[cfg(all(unix, target_vendor = "apple"))]
+impl<E> HasOnCrashReset for TimeoutExecutor<E> {
+    /// Disarm currently running timer
+    fn reset_on_crash(&self) {
+        unsafe {
+            let mut itimerval_zero: Itimerval = zeroed();
+            setitimer(ITIMER_REAL, &mut itimerval_zero, null_mut());
+        }
+    }
+}
+
+#[cfg(all(windows, feature = "std"))]
+impl<E> HasOnCrashReset for TimeoutExecutor<E> {
+    /// Deletes this timer queue
+    /// # Safety
+    /// Will dereference the given `tp_timer` pointer, unchecked.
+    fn reset_on_crash(&self) {
+        unsafe {
+            CloseThreadpoolTimer(self.tp_timer);
+        }
+    }
+}
+
 #[cfg(windows)]
 impl<E> Drop for TimeoutExecutor<E> {
     fn drop(&mut self) {
```

Comment:
As @tokatoka  suggested, this can completely be removed

Reply:
Yes, now I've found out that this is a bug! We should definitely remove this Drop impl